source/threads.tex

```diff
@@ -482,175 +1164,515 @@
+
+\pnum
+\sync The completion of the thread represented by \tcode{*this} synchronizes with\iref{intro.multithread}
+the corresponding successful
+\tcode{join()} return. \begin{note} Operations on
+\tcode{*this} are not synchronized. \end{note}
+
+\pnum
+\ensures The thread represented by \tcode{*this} has completed. \tcode{get_id() == id()}.
+
+\pnum
+\throws \tcode{system_error} when
+an exception is required\iref{thread.req.exception}.
+
+\pnum
+\errors
+\begin{itemize}
+\item \tcode{resource_deadlock_would_occur} --- if deadlock is detected or
+\tcode{get_id() == this_thread::\brk{}get_id()}.
+
+\item \tcode{no_such_process} --- if the thread is not valid.
+
+\item \tcode{invalid_argument} --- if the thread is not joinable.
+\end{itemize}
+\end{itemdescr}
+
+\indexlibrarymember{detach}{thread}%
+\begin{itemdecl}
+void detach();
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects The thread represented by \tcode{*this} continues execution without the calling thread
+blocking. When \tcode{detach()} returns, \tcode{*this} no longer represents the possibly continuing
+thread of execution. When the thread previously represented by \tcode{*this} ends execution, the
+implementation shall release any owned resources.
+
+\pnum\ensures \tcode{get_id() == id()}.
+
+\pnum\throws \tcode{system_error} when
+an exception is required\iref{thread.req.exception}.
+
+\pnum \errors
+\begin{itemize}
+\item \tcode{no_such_process} --- if the thread is not valid.
+\item \tcode{invalid_argument} --- if the thread is not joinable.
+\end{itemize}
+\end{itemdescr}
+
+\indexlibrarymember{get_id}{thread}%
+\begin{itemdecl}
+id get_id() const noexcept;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\returns A default constructed \tcode{id} object if \tcode{*this} does not represent a thread,
+otherwise \tcode{this_thread::get_id()} for the thread of execution represented by
+\tcode{*this}.
+\end{itemdescr}
+
+\rSec3[thread.thread.static]{Static members}
+
+\indexlibrarymember{hardware_concurrency}{thread}%
+\begin{itemdecl}
+unsigned hardware_concurrency() noexcept;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\returns The number of hardware thread contexts. \begin{note} This value should
+only be considered to be a hint. \end{note} If this value is not computable or
+well-defined, an implementation should return 0.
+\end{itemdescr}
+
+\rSec3[thread.thread.algorithm]{Specialized algorithms}
+
+\indexlibrarymember{swap}{thread}%
+\begin{itemdecl}
+void swap(thread& x, thread& y) noexcept;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum\effects As if by \tcode{x.swap(y)}.
+\end{itemdescr}
+
+\rSec2[thread.jthread.class]{Class \tcode{jthread}}
+
+\pnum
+The class \tcode{jthread} provides a mechanism
+to create a new thread of execution.
+The functionality is the same as for
+class \tcode{thread}\iref{thread.thread.class}
+with the additional ability to request that the thread stops
+and then joins the started thread.
```

Comment:
On the edit@ reflector I suggested changing this to:

> The class `jthread` provides a mechanism to create a new thread of execution. The functionality is the same as for class `thread` (32.4.2) with the additional abilities to provide a `stop_token` (32.3) to the new thread of execution, make stop requests, and automatically join.

Reply:
@jwakely Is this still an issue?  If so, can you please create an editorial/lwg issue for it?